core/src/main/java/org/apache/iceberg/util/PartitionSet.java

```diff
@@ -87,7 +87,7 @@ public boolean add(Pair<Integer, StructLike> pair) {
   }
 
   public boolean add(int specId, StructLike struct) {
-    Set<StructLike> partitionSet = partitionSetById.computeIfAbsent(specId,
+    StructLikeSet partitionSet = partitionSetById.computeIfAbsent(specId,
```

Comment:
Ok these were also safe before since we were always using StructLikeSet under the hood

Reply:
reverted

Reply:
Same here, I was just checking to make sure that there wasn't a bug here we needed to investigate